src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/UserController.kt

```diff
@@ -2,22 +2,46 @@ package org.vaccineimpact.api.app.controllers
 
 import org.vaccineimpact.api.app.ActionContext
 import org.vaccineimpact.api.app.controllers.endpoints.SecuredEndpoint
-import org.vaccineimpact.api.app.errors.UnknownObjectError
-import org.vaccineimpact.api.models.User
+import org.vaccineimpact.api.models.Scope
+import org.vaccineimpact.api.models.UserInterface
+import org.vaccineimpact.api.models.permissions.ReifiedPermission
+import org.vaccineimpact.api.models.permissions.RoleAssignment
 
 class UserController(context: ControllerContext) : AbstractController(context)
 {
     override val urlComponent = "/users"
-    override val endpoints = listOf(
-            SecuredEndpoint("/:username/", this::getUser, emptySet())
+    override val endpoints = listOf(SecuredEndpoint("/:username/", this::getUser, setOf("*/users.read"))
     )
 
-    fun getUser(context: ActionContext): User
+    fun getUser(context: ActionContext): UserInterface
     {
-        var userName = userName(context)
+        val userName = userName(context)
 
-        return repos.user().use { it.getUserByUsername(userName)  }
+        val roleReadingPermissions = context.permissions.filter { p -> p.name == "roles.read" }
+
+        if (!roleReadingPermissions.any())
+            return repos.user().use { it.getUserByUsername(userName) }
+
+        val userWithRoles = repos.user().use { it.getUserWithRolesByUsername(userName) }
+
+        userWithRoles.roles = userWithRoles.roles.filter { r ->
+            roleReadingPermissions.any {
+                p ->  p.scope.encompasses(parseScope(r))
+            }
+        }
+
+        return userWithRoles
     }
 
+    private fun parseScope(role: RoleAssignment) =
```

Comment:
It might be good to put this in `Scope`'s companion object. e.g. `fun fromRoleAssignment` so it can be reused. Also, rather than invoking `Scope.parse`, you may as well construct a `Scope.Specific` object directly.